awssqs/src/main/java/zipkin/reporter/awssqs/AwsBufferedSqsSender.java

```diff
@@ -0,0 +1,235 @@
+    public abstract Builder maxRequests(int maxRequests);
+    public abstract Builder maxQueuedRequests(int maxQueuedRequests);
+
+
+    public abstract Builder encoding(Encoding encoding);
+
+    /**
+     * Maximum size of a message. SQS max message size is 256KB including attributes.
+     * Default 256 less 30 bytes for message keys.
+     *
+     * @param messageMaxBytes
+     * @return
+     */
+    public abstract Builder messageMaxBytes(int messageMaxBytes);
+
+    public final AwsBufferedSqsSender build() {
+      return autoBuild();
+    }
+
+    abstract AwsBufferedSqsSender autoBuild();
+
+    Builder() {}
+  }
+
+  public Builder toBuilder() {
+    return new AutoValue_AwsBufferedSqsSender.Builder(this);
+  }
+
+  abstract String queueUrl();
+  @Nullable abstract String awsAccessKeyId();
+  @Nullable abstract String awsSecretAccessKey();
+  @Nullable abstract String awsRoleName();
+  @Nullable abstract String awsServiceId();
+  abstract boolean compressionEnabled();
+  abstract int maxRequests();
+  abstract int maxQueuedRequests();
+
+  /** close is typically called from a different thread */
+  private AtomicBoolean closeCalled = new AtomicBoolean(false);
+
+
+  @Override public CheckResult check() {
+    try {
+      List<String> queueUrls = get().listQueues().getQueueUrls();
+      if (queueUrls.contains(queueUrl())) {
+        return CheckResult.OK;
+      } else {
+        return CheckResult.failed(new IllegalStateException(String.format("queue url not found %s", queueUrl())));
+      }
+    } catch (Exception e) {
+      return CheckResult.failed(e);
+    }
+  }
+
+  @Override protected AmazonSQSBufferedAsyncClient compute() {
+    // bound the executor so that we get consistent performance
+    ThreadPoolExecutor dispatchExecutor =
+        new ThreadPoolExecutor(0, maxRequests(), 60, TimeUnit.SECONDS,
+            new ArrayBlockingQueue<>(maxQueuedRequests()),
+            threadFactory("AwsBufferedSqsSender Dispatcher", false));
+
+    AWSCredentialsProvider credentialsProvider =
+        credentialsProvider(awsAccessKeyId(), awsSecretAccessKey(), awsServiceId(), awsRoleName());
+
+    return new AmazonSQSBufferedAsyncClient(
+        new AmazonSQSAsyncClient(credentialsProvider, dispatchExecutor));
+  }
+
+  @Override public int messageSizeInBytes(List<byte[]> encodedSpans) {
+    return encoding().listSizeInBytes(encodedSpans);
+  }
+
+  @Override public void sendSpans(List<byte[]> encodedSpans, Callback callback) {
+    if (closeCalled.get()) throw new IllegalStateException("closed");
+
+    checkNotNull(encodedSpans, "list of encoded spans must not be null");
+
+    for (byte[] span : encodedSpans) {
+      get().sendMessageAsync(new SendMessageRequest(queueUrl(), "z")
+              .addMessageAttributesEntry("z", new MessageAttributeValue()
+                  .withDataType("Binary." + encoding().name())
+                  .withBinaryValue(ByteBuffer.wrap(span))), new AsyncHandlerAdapter(callback));
+    }
```

Comment:
Could you paste an example of what this payload looks like on the receiving end or in the SQS console. I'm trying to decide what I need an SNS payload to look like so that the collector will be able to distinguish between the 2 payload types.